### PR TITLE
Doston

### PR DESCRIPTION
**Pull request by dostonbekqosimov**
All apis except for 8 for Channel finished

### CODE REVIEW

File: src/main/java/dasturlash/uz/dto/response/ChannelMediaDTO.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package dasturlash.uz.dto.response;
+
+import lombok.Data;
+
+@Data
+public class ChannelMediaDTO {
```
Comment (dostonbekqosimov) [inline src/main/java/dasturlash/uz/dto/response/ChannelMediaDTO.java:6]: This for sending url of photo and banner of Channel

File: src/main/java/dasturlash/uz/dto/response/ChannelResponseDTO.java [hunk 2]
```diff
@@ -20,11 +20,11 @@ public class ChannelResponseDTO {
 
     private String description;
 
-    private String photoId;
+    private ChannelMediaDTO photo;
```
Comment (dostonbekqosimov) [inline src/main/java/dasturlash/uz/dto/response/ChannelResponseDTO.java:23]: Bu avval String edi endi yuqorida, Channelni banner va photo larini open qilish uchun url ni o'zida saqlaydigan class ga o'zgartirildi

File: src/main/java/dasturlash/uz/dto/response/ChannelResponseDTO.java [hunk 3]
```diff
@@ -20,11 +20,11 @@ public class ChannelResponseDTO {
...
 
     private Long profileId;
 
-    private String bannerId;
+    private ChannelMediaDTO banner;
```
Comment (dostonbekqosimov) [inline src/main/java/dasturlash/uz/dto/response/ChannelResponseDTO.java:27]: Bu avval String edi endi yuqorida, Channelni banner va photo larini open qilish uchun url ni o'zida saqlaydigan class ga o'zgartirildi

File: src/main/java/dasturlash/uz/repository/ChannelRepository.java [hunk 4]
```diff
@@ -16,4 +18,6 @@ public interface ChannelRepository extends CrudRepository<Channel, String> {
 
     @Query("select c from Channel c where lower(c.handle) like lower(concat('%', :handle, '%')) and c.visible = true")
     List<Channel> findByHandle(@Param("handle")String channelHandle);
+
+    Page<Channel> findAllByVisibleTrue(Pageable pageable);
```
Comment (dostonbekqosimov) [inline src/main/java/dasturlash/uz/repository/ChannelRepository.java:22]: Bu Channel larni listini pagination bilan olish uchun qo'shildi

File: src/main/java/dasturlash/uz/security/SpringSecurityUtil.java [hunk 5]
```diff
@@ -18,4 +19,16 @@ public static Long getCurrentUserId() {
         }
         throw new AppBadRequestException("http://localhost:8090/api/auth/login");
     }
+
+    public static CustomUserDetails getCurrentEntity() {
```
Comment (dostonbekqosimov) [inline src/main/java/dasturlash/uz/security/SpringSecurityUtil.java:23]: Tizimga kirgan pfofileni malumotlarini olish uchun qo'shildi(asosan shu classni ichida ishlatilinadi)

File: src/main/java/dasturlash/uz/security/SpringSecurityUtil.java [hunk 6]
```diff
@@ -18,4 +19,16 @@ public static Long getCurrentUserId() {
...
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        CustomUserDetails userDetail = (CustomUserDetails) authentication.getPrincipal();
+
+        return userDetail;
+    }
+
+    public static ProfileRole getCurrentUserRole() {
```
Comment (dostonbekqosimov) [inline src/main/java/dasturlash/uz/security/SpringSecurityUtil.java:30]: Profile ni role olinyabdi

File: src/main/java/dasturlash/uz/service/AttachService.java [hunk 7]
```diff
@@ -286,6 +286,15 @@ private String getPath(Attach entity) {
         return folderName + "/" + entity.getPath() + "/" + entity.getId();
     }
 
+    public ChannelMediaDTO getUrlOfMedia(String attachId) {
```
Comment (dostonbekqosimov) [inline src/main/java/dasturlash/uz/service/AttachService.java:289]: url yasalyabdi (photo and banner of Channel)

**Review by dostonbekqosimov — COMMENTED** [review]
gg

**Review by saidovdiyorbek — COMMENTED** [review]
gg

**Pull request merged**